internal/reader/rmn_home.go

```diff
@@ -249,25 +244,29 @@ func validate(config VersionedConfigWithDigest) error {
 
 func convertOnChainConfigToRMNHomeChainConfig(
 	lggr logger.Logger,
-	versionedConfigWithDigests []VersionedConfigWithDigest,
+	primaryConfig VersionedConfig,
+	secondaryConfig VersionedConfig,
 ) map[cciptypes.Bytes32]rmntypes.HomeConfig {
-	if len(versionedConfigWithDigests) == 0 {
-		lggr.Warnw("no on chain RMNHomeConfigs found")
+	if primaryConfig.ConfigDigest.IsEmpty() && secondaryConfig.ConfigDigest.IsEmpty() {
+		lggr.Warnw("no on chain RMNHomeConfigs found, both digests are empty")
```

Comment:
If you return an error here you can check that error in `fetchAndSetHomeChainConfigs` instead of having the duplicate check.